### DISCUSSION TITLE
recover response_model from output annotations

### DISCUSSION THREAD

**Discussion by AM-I-Human** [fastapi/fastapi#6355] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI

app = FastAPI()


@router.post("/have_output_example_and_schema", response_model=list)
async def test(input: int-> list:
    return []


@router.post("/have_not_output_example_and_schema")
async def test_2(input: int-> list:
    return []
```

### Description

I think it will be nice if the type in response_model (in the app decorators) will be infered from the output annotation of the function.
By doing so you can avoid some documentation-code inconsistency.

### Wanted Solution

When response_model is None it will try to use the annotation one, es by using `self.response_model= response_model or get_output_annotation(fn)`

```python
@router.post("/have_not_output_example_and_schema")
async def test_2(input: int-> list:
    return []
```
Will be the same as

```python
@router.post("/have_not_output_example_and_schema",response_model=list)
async def test_2(input: int-> list:
    return []
```

### Wanted Code

```python
from fastapi import FastAPI

app = FastAPI()


@router.post("/have_output_example_and_schema")
async def test(input: int-> list:
    return []


@router.post("/have_not_output_example_and_schema")
async def test_2(input: int-> list:
    return []
```

### Operating System

Windows

### Operating System Details

Windows 11

### FastAPI Version

0.75.0

### Python Version

Python 3.10.4

**Comment by chbndrhnns** [comment, marked as answer]
This has been discarded earlier, like in https://github.com/tiangolo/fastapi/issues/101

**Comment by AM-I-Human** [comment]
Ah! I missed that, thanks.

**Comment by tiangolo** [comment]
Thanks for the help here @chbndrhnns! 👏  🙇 

Thanks for reporting back and closing the issue @AM-I-Human  👍

This is now supported (optionally) in the latest versions. :rocket: 

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
